src/reconcile/rec_visibility.c

```diff
@@ -216,8 +216,11 @@ __rec_need_save_upd(
     if (F_ISSET(r, WT_REC_CHECKPOINT) && upd_select->upd == NULL)
         return (false);
 
-    return (!__wt_txn_tw_stop_visible_all(session, &upd_select->tw) &&
-      !__wt_txn_tw_start_visible_all(session, &upd_select->tw));
+    if (WT_TIME_WINDOW_HAS_STOP(&upd_select->tw))
```

Comment:
Why we need this?

Reply:
The older approach is if there is no stop timestamp it always returns false. The problem is when there is a stop timestamp, we should check both the start and stop should be globally visible, but the old check will pass if any one of them is globally visible. So I changed the check to check for both stop and start only if the stop is present.

Reply:
Should we only need to check the stop timestamp to be globally visible if there is a stop timestamp?

Reply:
If there is a stop timestamp and it is not globally visible, we still cannot discard the update chain.

Reply:
Earlier by default RTS skips the pages that don't have the timestamp information, now with this error, I changed it to don't skip those pages, but it is better to fix the behaviour of reconciliation to not remove the unstable updates from the page.

Also, I changed the condition to check only to global visibility of the stop timestamp.